iii_account/iii_account.py

```diff
@@ -39,10 +41,8 @@ def login(self):
                'url': None}
         url = self.opac_url + 'patroninfo'
         rsp = self.session.post(url, data=payload, allow_redirects=True)
-        content = rsp.content
-        doc = pq(rsp.content)
-        # Span.login_error will also appear in content if login fails.
-        login_error = doc('span.login_error').val()
+        doc = BeautifulSoup( rsp.content.decode('utf-8') )
+        login_error = doc.find( "span", class_="login_error" )  # <span class="login_error"> exists if login fails
         if (login_error):
             raise Exception("Login failed.")
```

Comment:
At some point we might want to raise a more explicit Exception so that the client can display appropriate error messages or change workflow.